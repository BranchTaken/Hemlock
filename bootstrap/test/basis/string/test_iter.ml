open! Basis.Rudiments
open! Basis
open String

let test () =
  let test_iter s = begin
    File.Fmt.stdout
    |> Basis.Fmt.fmt "iter "
    |> pp s
    |> Basis.Fmt.fmt " ->"
    |> ignore;
    let () = iter s ~f:(fun cp ->
      File.Fmt.stdout
      |> Basis.Fmt.fmt " "
      |> Basis.Fmt.fmt (of_codepoint cp)
      |> ignore
    ) in
    File.Fmt.stdout |> Basis.Fmt.fmt "\n" |> ignore
  end in
  let strs = [
    "";
    "abcde";
  ] in
  List.iter strs ~f:test_iter

let _ = test ()
