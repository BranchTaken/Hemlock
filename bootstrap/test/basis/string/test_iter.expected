iter "" ->
iter "abcde" -> a b c d e
